### ISSUE TITLE
'ValueError: Unrecognised status PASS' when running update-wpt

### ISSUE THREAD

**Issue by frewsxcv** [servo/servo#8414]
```
coreyf@frewbook-pro ~/D/r/servo (html-hr-element-color-attribute) [1]> ./mach update-wpt /tmp/servo.log
Warning: Duplicate heading /html/semantics/text-level-semantics/the-wbr-element/wbr-element.html
Error running mach:

    ['update-wpt', '/tmp/servo.log']

The error occurred in code that was called by the mach command. This is either
a bug in the called code itself or in the way that mach is calling it.

You should consider filing a bug for this issue.

If filing a bug, please include the full output of mach, including this error
message.

The details of the failure are as follows:

ValueError: Unrecognised status PASS

  File "/Users/coreyf/Development/rust/servo/python/servo/testing_commands.py", line 300, in update_wpt
    return run_globals["update_tests"](**kwargs)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/update.py", line 25, in update_tests
    rv = update.run_update(logger, **kwargs)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/update/__init__.py", line 25, in run_update
    return updater.run()
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/update/update.py", line 149, in run
    rv = update_runner.run()
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/update/base.py", line 65, in run
    rv = step(self.logger).run(step_index, self.state)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/update/base.py", line 36, in run
    self.create(state)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/update/update.py", line 95, in create
    runner.run()
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/update/base.py", line 65, in run
    rv = step(self.logger).run(step_index, self.state)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/update/base.py", line 36, in run
    self.create(state)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/update/metadata.py", line 27, in create
    sync_root=sync_root)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/metadata.py", line 54, in update_expected
    ignore_existing=ignore_existing)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/metadata.py", line 145, in update_from_logs
    updater.update_from_log(f)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/metadata.py", line 225, in update_from_log
    reader.each_log(log_reader, self.action_map)
  File "/Users/coreyf/Development/rust/servo/python/_virtualenv/lib/python2.7/site-packages/mozlog/reader.py", line 48, in each_log
    action_map[item["action"]](item)
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/metadata.py", line 281, in test_end
    data.get("message"))
  File "/Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/wpttest.py", line 19, in __init__
    raise ValueError("Unrecognised status %s" % status)
```

**Comment by jgraham** [comment]
Got anywhere to put the failing input?

**Comment by frewsxcv** [comment]
http://froosky.rwell.org/present-for-jgraham/servo.log

Log generated via `./mach test-wpt --log-raw=/tmp/servo.log --release` with [this branch](https://github.com/frewsxcv/servo/tree/html-hr-element-color-attribute)

**Comment by jdm** [comment]
Probably http://mxr.mozilla.org/servo/source/tests/wpt/metadata-css/css21_dev/html4/block-in-inline-remove-002.htm.ini#3 and http://mxr.mozilla.org/servo/source/tests/wpt/metadata-css/css21_dev/html4/block-in-inline-remove-000.htm.ini#3 .

**Comment by frewsxcv** [comment]
I removed those files and still get the same errors

**Comment by jgraham** [comment]
Oops, sorry I rather dropped this. I don't think that should be the issue; from the backtrace it looks like you're getting a PASS for something where that isn't possible i.e. in the test_end of a testharness.js test. I suspect some .ini file has `type: testharness` rather than `type: reftest`. If you go up one level in the stack and look at `test.name` I think that will tell you the filename of the broken manifest file.

**Comment by frewsxcv** [comment]
```
> /Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/wpttest.py(20)__init__()
-> raise ValueError("Unrecognised status %s" % status)
(Pdb) vars()
{'status': u'PASS', 'extra': None, 'self': <wptrunner.wpttest.TestharnessResult object at 0x1259d0e50>, 'expected': None, 'message': None, 'pdb': <module 'pdb' from '/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/pdb.pyc'>}
```
```
(Pdb) up
> /Users/coreyf/Development/rust/servo/tests/wpt/harness/wptrunner/metadata.py(282)test_end()
-> data.get("message"))
(Pdb) vars()
{'test': <ManifestItem tex_image_2d_canvas_no_context.html>, 'self': <wptrunner.metadata.ExpectedUpdater object at 0x10152d290>, 'test_id': u'/_mozilla/mozilla/webgl/tex_image_2d_canvas_no_context.html', 'data': {u'status': u'PASS', u'thread': u'Thread-TestrunnerManager-13', u'pid': 12318, u'source': u'web-platform-tests', u'time': 1447392815265, u'test': u'/_mozilla/mozilla/webgl/tex_image_2d_canvas_no_context.html', u'action': u'test_end'}, 'test_cls': <class 'wptrunner.wpttest.TestharnessTest'>}
```
```
(Pdb) up
> /Users/coreyf/Development/rust/servo/python/_virtualenv/lib/python2.7/site-packages/mozlog/reader.py(48)each_log()
-> action_map[item["action"]](item)
(Pdb) vars()
{'item': {u'status': u'PASS', u'thread': u'Thread-TestrunnerManager-13', u'pid': 12318, u'source': u'web-platform-tests', u'time': 1447392815265, u'test': u'/_mozilla/mozilla/webgl/tex_image_2d_canvas_no_context.html', u'action': u'test_end'}, 'log_iter': <generator object read at 0x1011cec80>, 'action_map': {'test_end': <bound method ExpectedUpdater.test_end of <wptrunner.metadata.ExpectedUpdater object at 0x10152d290>>, 'suite_start': <bound method ExpectedUpdater.suite_start of <wptrunner.metadata.ExpectedUpdater object at 0x10152d290>>, 'test_start': <bound method ExpectedUpdater.test_start of <wptrunner.metadata.ExpectedUpdater object at 0x10152d290>>, 'test_status': <bound method ExpectedUpdater.test_status of <wptrunner.metadata.ExpectedUpdater object at 0x10152d290>>}}
```

**Comment by jgraham** [comment]
If you look at the metadata file. `tests/wpt/mozilla/meta/mozilla/webgl/tex_image_2d_canvas_no_context.html.ini` claims to be `type: testharness` whereas in fact the corresponding test file is clearly a reftest. You just need to fix the ini file to be right.

**Comment by frewsxcv** [comment]
8fff34e2e78137b5117ab0db86f9aadd56cc50e5

**Comment by frewsxcv** [comment]
> claims to be type: testharness whereas in fact the corresponding test file is clearly a reftest

I feel like this should cause the test to fail if it's not set to the right type

**Comment by jgraham** [comment]
Yeah it could either check this when loading the tests (easy to fix) or not require this in the ini file at all (harder to fix).

**Comment by frewsxcv** [comment]
This was fixed in #8537. 

> Yeah it could either check this when loading the tests (easy to fix) or not require this in the ini file at all (harder to fix).

Opened a new issue for this: #8540
